website/docs/reference/resource-configs/hard-deletes.md

```diff
@@ -0,0 +1,111 @@
+---
+title: hard_deletes
+resource_types: [snapshots]
+description: "Use the `hard_deletes` config to control how deleted rows are tracked in your snapshot table."
+datatype: "{<dictionary>}"
+default_value: {ignore}
+id: "hard-deletes"
+sidebar_label: "hard_deletes"
+---
+
+Available from dbt v1.9 or with [Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless) dbt Cloud.
+
+
+<File name='snapshots/schema.yml'>
+
+```yaml
+snapshots:
+  - name: <snapshot_name>
+    config:
+      hard_deletes: 'ignore' | 'invalidate' | 'new_record'
+```
+</File>
+
+<File name='dbt_project.yml'>
+
+```yml
+snapshots:
+  [<resource-path>](/reference/resource-configs/resource-path):
+    +hard_deletes: "ignore" | "invalidate" | "new_record"
+```
+
+</File>
+
+<File name='snapshots/<filename>.sql'>
+
+```sql
+{{
+    config(
+        unique_key='id',
+        strategy='timestamp',
+        updated_at='updated_at',
+        hard_deletes='ignore' | 'invalidate' | 'new_record'
+    )
+}}
+```
+
+</File>
+
+
+## Description
+
+The `hard_deletes` config gives you more control on how to handle deleted rows from the source. Supported options are `ignore` (default), `invalidate` (replaces the legacy `invalidate_hard_deletes=true`), and `new_record`.
+
+import HardDeletes from '/snippets/_hard-deletes.md';
+
+<HardDeletes />
+
+:::warning
+
+If you're updating an existing snapshot to use the `hard_deletes` config, dbt _will not_ handle migrations automatically. We recommend either only using these settings for net-new snapshots, or arranging an update of pre-existing tables before enabling this setting.
+:::
```

Comment:
Link to the "Snapshot configuration migration" section added in https://github.com/dbt-labs/docs.getdbt.com/pull/6521?